temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -508,6 +534,14 @@ private void handleCommandEvent(HistoryEvent event) {
     if (handleLocalActivityMarker(event)) {
       return;
     }
+    if (shouldSkipUpsertVersionSA) {
+      if (handleNonMatchingUpsertSearchAttribute(event)) {
+        shouldSkipUpsertVersionSA = false;
```

Comment:
Is this bool only valid for this one event? Does setting it back to false here make a difference, or is it just to be safe?

Reply:
Yes it is only valid for one event